controllers/alphacontrollers/outboundwebhook_controller.go

```diff
@@ -0,0 +1,453 @@
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the OutboundWebhook object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.4/pkg/reconcile
+var (
+	outboundWebhookFinalizerName = "outbound-webhook.coralogix.com/finalizer"
+)
+
+func (r *OutboundWebhookReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var (
+		resultError = ctrl.Result{RequeueAfter: 40}
+		resultOk    = ctrl.Result{RequeueAfter: defaultErrRequeuePeriod}
+		err         error
+	)
+
+	log := log.FromContext(ctx).WithValues(
+		"outboundWebhook", req.NamespacedName.Name,
+		"namespace", req.NamespacedName.Namespace,
+	)
+
+	outboundWebhook := &coralogixv1alpha1.OutboundWebhook{}
+	if err = r.Client.Get(ctx, req.NamespacedName, outboundWebhook); err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		return resultError, err
+	}
+
+	if ptr.Deref(outboundWebhook.Status.ID, "") == "" {
+		err = r.create(ctx, log, outboundWebhook)
+		if err != nil {
+			log.Error(err, "Error on creating outbound-webhook")
+			return resultError, err
+		}
+		return resultOk, nil
+	}
+
+	if !outboundWebhook.ObjectMeta.DeletionTimestamp.IsZero() {
+		err = r.delete(ctx, log, outboundWebhook)
+		if err != nil {
+			log.Error(err, "Error on deleting outbound-webhook")
+			return resultError, err
+		}
+		return resultOk, nil
+	}
+
+	err = r.update(ctx, log, outboundWebhook)
+	if err != nil {
+		log.Error(err, "Error on updating outbound-webhook")
+		return resultError, err
+	}
+
+	return resultOk, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *OutboundWebhookReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&coralogixv1alpha1.OutboundWebhook{}).
+		Complete(r)
+}
+
+func (r *OutboundWebhookReconciler) create(ctx context.Context, log logr.Logger, webhook *coralogixv1alpha1.OutboundWebhook) error {
+	createRequest, err := webhook.ExtractCreateOutboundWebhookRequest()
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Error to extract create-request out of the outbound-webhook -\n%v", webhook))
+		return err
+	}
+
+	log.V(1).Info(fmt.Sprintf("Creating outbound-webhook-\n%s", protojson.Format(createRequest)))
+	createResponse, err := r.OutboundWebhooksClient.CreateOutboundWebhook(ctx, createRequest)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Received an error while creating outbound-webhook -\n%s", protojson.Format(createRequest)))
```

Comment:
Let's only log on the higher function, here we should always return an error message.

```suggestion
return fmt.Errorf("received an error while trying to createoutbound-webhook CRD: %w", err)
```

Since we're adding the following code on the reconcile function

```
log := log.FromContext(ctx).WithValues(
		"outboundWebhook", req.NamespacedName.Name,
		"namespace", req.NamespacedName.Namespace,
	)
```

All logs will contain the webhook name and name space

Reply:
This is more informative, because it shows the call for the backend. This way we can easily track where is the issue.